libevmasm/RuleList.h

```diff
@@ -577,29 +589,30 @@ std::vector<SimplificationRule<Pattern>> simplificationRuleListPart9(
 	Pattern Z
 )
 {
+	using Word = typename Pattern::Word;
 	std::vector<SimplificationRule<Pattern>> rules;
 
-	u256 const mask = (u256(1) << 160) - 1;
+	Word const mask = (Word(1) << 160) - 1;
```

Comment:
Also put this whole block into ``if (Pattern::WordSize > 160)``?
It might actually be possible to use ``>= 160``, but then we need to verify that the mask is correct for ``= 160``.